base/linalg/triangular.jl

```diff
@@ -1741,188 +1741,187 @@ powm(A::LowerTriangular, p::Real) = powm(A.', p::Real).'
-        else
-            logAk = log(Ak)
-            logAkp1 = log(Akp1)
-            w = atanh((Akp1 - Ak)/(Akp1 + Ak)) + im*pi*ceil((imag(logAkp1-logAk)-pi)/(2*pi))
-            dd = 2 * exp(p*(logAk+logAkp1)/2) * sinh(p*w) / (Akp1 - Ak)
-            A[k,k+1] = A0[k,k+1] * dd
+        if flag != 2
+            A, mm, its = sqrtm_dbp(A)
+            if any(isnan.(A))
+                return UpperTriangular(A)
+            end
+            AmI = A - I
+            s += 1
+            if s == 1
+                z0 = AmI
+            elseif s == 2
+                R = I + A
+            else
+                R = R * (I + A)
+            end
         end
     end
 
-    # Compute accurate diagonal of T
-    for i = 1:n
-        a = A0[i,i]
-        if s == 0
-            r = a - 1
-        end
-        s0 = s
-        if angle(a) >= pi / 2
-            a = sqrt(a)
-            s0 = s - 1
-        end
-        z0 = a - 1
-        a = sqrt(a)
-        r = 1 + a
-        for j = 1:s0-1
-            a = sqrt(a)
-            r = r * (1 + a)
-        end
-        A[i,i] = z0 / r
+    if s >= 2
+        R = z0 / R
+    else
+        R = AmI
     end
 
     # Get the Gauss-Legendre quadrature points and weights
-    R = zeros(Float64, m, m)
+    v = zeros(Float64, m, m)
     for i = 1:m - 1
-        R[i,i+1] = i / sqrt((2 * i)^2 - 1)
-        R[i+1,i] = R[i,i+1]
+        v[i, i+1] = i / sqrt((2 * i)^2 - 1)
+        v[i+1, i] = v[i, i+1]
     end
-    x,V = eig(R)
-    w = Vector{Float64}(m)
+
+    x, V = eig(v)
+    w = Vector(m)
+
     for i = 1:m
         x[i] = (x[i] + 1) / 2
-        w[i] = V[1,i]^2
+        w[i] = V[1, i]^2
     end
 
-    # Compute the Padé approximation
-    Y = zeros(T, n, n)
+    # Compute the Padé approximation to matrix logarithm
+    Y = zeros(n, n)
     for k = 1:m
-        Y = Y + w[k] * (A / (x[k] * A + I))
+        Y = Y + w[k] * (R / (x[k] * R + I))
     end
 
     # Scale back
     scale!(2^s, Y)
 
-    # Compute accurate diagonal and superdiagonal of log(T)
-    for k = 1:n-1
-        Ak = A0[k,k]
-        Akp1 = A0[k+1,k+1]
-        logAk = log(Ak)
-        logAkp1 = log(Akp1)
-        Y[k,k] = logAk
-        Y[k+1,k+1] = logAkp1
-        if Ak == Akp1
-            Y[k,k+1] = A0[k,k+1] / Ak
-        elseif 2 * abs(Ak) < abs(Akp1) || 2 * abs(Akp1) < abs(Ak)
-            Y[k,k+1] = A0[k,k+1] * (logAkp1 - logAk) / (Akp1 - Ak)
-        else
-            w = atanh((Akp1 - Ak)/(Akp1 + Ak) + im*pi*(ceil((imag(logAkp1-logAk) - pi)/(2*pi))))
-            Y[k,k+1] = 2 * A0[k,k+1] * w / (Akp1 - Ak)
-        end
-    end
-
     return UpperTriangular(Y)
 end
 logm(A::LowerTriangular) = logm(A.').'
 
 # Auxiliary functions for logm and matrix power
 
+# Matrix square root by product form of Denman-Beavers iteration.
+#   computes the principal square root X
+#   of the matrix A using the product form of the Denman-Beavers
+#   iteration. The matrix M tends to I.
+#       Copyright (c) 2008 by N. J. Higham
```

Comment:
he's been very nice before about granting permission to relicense derived translations of his implementations, did a previous instance of that also cover this bit of code?